autokeras/task.py

```diff
@@ -169,16 +171,106 @@ def __init__(self,
 
 class SupervisedStructuredDataPipeline(auto_model.AutoModel):
 
-    def __init__(self, outputs, **kwargs):
-        super().__init__(inputs=node.StructuredDataInput(),
+    def __init__(self, outputs, column_names, column_types, **kwargs):
+        # TODO: support customized column_types.
+        inputs = node.StructuredDataInput()
+        if column_types:
+            inputs.column_types = column_types
+            if column_names is None:
+                raise ValueError('Column names must be specified.')
```

Comment:
We cannot tell that the column names are not specified in the init function since they might be in the csv file, which can only be known in fit. So move this check to fit.